domain/src/main/java/org/oppia/android/domain/question/QuestionAssessmentProgress.kt

```diff
@@ -47,6 +47,22 @@ internal class QuestionAssessmentProgress {
     questionSessionMetrics[currentQuestionIndex].didViewSolution = true
   }
 
+  /**
+   * Called every time the user submits an answer to track how many answers the user submitted per question, along with
+   * any misconceptions.
```

Comment:
Method docs should always start with the action being taken, then why. E.g.:

```suggestion
   * Tracks how many answers the user submits for each question, along with any misconceptions. This
   * is expected to be called for every answer submitted.
```

Reply:
Fixed the documentation for this method as well as the previous tracking methods I added. Thanks!